src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -302,7 +302,10 @@ func (enc *encoder) LastEncoded() (ts.Datapoint, error) {
 		return ts.Datapoint{}, errNoEncodedDatapoints
 	}
 
-	result := ts.Datapoint{Timestamp: enc.tsEncoderState.PrevTime}
+	result := ts.Datapoint{
+		Timestamp:      enc.tsEncoderState.PrevTime,
+		TimestampNanos: xtime.ToUnixNano(enc.tsEncoderState.PrevTime),
```

Comment:
Hm, should we move to add a method called `NewDatapoint(t time.Time, v float64)` and then always add the fields ourselves?

Otherwise I think other callsites might miss this as well. 

We can change the `Timestamp` and `Value` to unexported and add methods to access them (which will likely get inlined by Go). (i.e. `Timestamp() time.Time` and `TimestampNanos() int64` and `Value() float64`

Reply:
Yeah, I was thinking that... I think there are a lot of places this will touch though; ok to do it as a follow up?